Python3 installation

rajatagarwal-ibm opened the pull request:
### Description

<!--- Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it. --->

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

vburckhardt commented on docs/contribute-module.md:
```diff
@@ -13,12 +13,12 @@ Follow these steps to create or update a module.
 
 - If you're running Microsoft Windows, set up Windows Subsystem for Linux [WSL](https://ubuntu.com/wsl) and run commands within WSL.
-- Make sure that you have a current version of Python 3 and pip.
+- Make sure that you have a current version of Python 3 and pip ([Python installation instructions](python3-mac-installation.md)).
```
I'd call out the mac instructions in a sub bullet as not all users are on mac

vburckhardt replied:
Also highlight that the python version provided out of the box on mac is not suitable

vburckhardt commented on docs/python3-mac-installation.md:
```diff
@@ -0,0 +1,13 @@
+# Latest Python3 installation on macOS
+
+_Note:_ This guide does not cover `homebrew` based Python3 installation.
```
Highlight the reason for those steps: the python version bundled with mac out the box is not suitable

vburckhardt reviewed (commented):
Thanks @rajatagarwal-ibm - a few comments

vburckhardt reviewed (approved):
lgtm

vburckhardt commented:
Adding @SirSpidey for final say

SirSpidey commented on docs/contribute-module.md:
```diff
@@ -13,12 +13,13 @@ Follow these steps to create or update a module.
 - If you're running Microsoft Windows, set up Windows Subsystem for Linux [WSL](https://ubuntu.com/wsl) and run commands within WSL.
-- Make sure that you have a current version of Python 3 and pip.
+- Make sure that you have a current version of Python 3 and pip. 
+  - _Notes for the macOS users:_ The out-of-the-box Python (Python 2.x) which comes with macOS is not suitable. The installation instructions for the latest Python 3.x version on macOS can be found [here](python3-mac-installation.md).
```
Do we need to tell people this? Have we run into issues? Python 2 has been deprecated for years. 

If you still think a notice is required, let's just tell them how to check.

```suggestion

    To check your installed version, run `python --version`.
```

SirSpidey commented on docs/contribute-module.md:
```diff
@@ -13,12 +13,13 @@ Follow these steps to create or update a module.
 ## Request a repo
 
-?> **Tip:** If you're updating an existing module, skip to the next step.
+**Tip:** If you're updating an existing module, skip to the next step.
```
this needs to return to the previous syntax: `?>`. It creates the callout/admonition style.

SirSpidey commented on docs/python3-mac-installation.md:
```diff
@@ -0,0 +1,15 @@
+1. Navigate to the [official Python download page for macOS](https://www.python.org/downloads/macos/).
+2. Click on the latest stable Python release.
+3. Under the files tab find the list of installers, click and download macOS appropriate installer locally.
+   ![installation_list.png](images/installation_list.png)
```
We don't include screenshots unless the text can't make it clear how to do something in the UI. This is because it creates more maintenance (and it tougher if translation is done).

I don't see the reason for including this.

SirSpidey commented on docs/python3-mac-installation.md:
```diff
@@ -0,0 +1,15 @@
+# Latest Python3 installation on macOS
```
What's the problem we're solving? Do people not know how to install Python? If we agree we need something, we can just include a link, as in, 

Install the [version of Python](https://www.python.org/downloads/) for your OS.

SirSpidey commented on docs/local-dev-setup.md:
```diff
@@ -2,11 +2,12 @@
 Follow these steps to set up your local development environment for working with the `terraform-ibm-modules` project.
 
-?> **Tip**: For regular maintenance tasks, see [Regular developer tasks](dev-maintenance.md).
+**Tip**: For regular maintenance tasks, see [Regular developer tasks](dev-maintenance.md).
```
this needs to return to the previous syntax: `?>`. It creates the callout/admonition style.

SirSpidey reviewed (changes requested):
Why do we need this update? There's no description and no linked issue that I saw. 

Python 2 has been deprecated for years. Apple pulled Python about 2 years ago with MacOS Monterey 12.3 and we're now at 13.4.

vburckhardt commented:
@toddgiguere @rajatagarwal-ibm could you chime in on @SirSpidey questions. It seems the consensus on the call was that mac users need to perform an extra step to install the right python version on their mac to get pre-commit to install. Not a mac user so I do not have the full visibility and rely on you.

rajatagarwal-ibm commented:
I have updated the PR and made it much cleaner. IMO the extensive documentation on python3 installation is not needed and doesn't cover all the scenarios. Hence will be harder to maintain. I have added a statement to see the current Python version as @SirSpidey  has pointed out. In the existing documentation, it is already covered that we need Python3 for the modules to work.

SirSpidey reviewed (approved):
works for me. Thanks.

ocofaigh reviewed (commented):
@rajatagarwal-ibm @SirSpidey This command may not work for everyone. For me I get:
```
% python --version
zsh: command not found: python
```

I need to run `python3 --version` to see the version. So I think this needs to be updated

SirSpidey commented:
@ocofaigh @rajatagarwal-ibm 
I think this is an issue with macs that had python2 installed and now don't. Because my new mac didn't have python installed, both `python3` and `python` work for me. 

Is it a bad thing that someone who sees that error and forgets that they need to use `python3` reinstalls python (which would get them 3)?

rajatagarwal-ibm commented:
@ocofaigh @SirSpidey How about we have it like this?

`python3 --version` or `python --version`

It will cover most scenarios.

SirSpidey commented:
@rajatagarwal-ibm I think Conall is correct. If you have python3 installed, `python3 --version` will work. So how about just updating to that?

rajatagarwal-ibm commented:
Alright let me create a quick PR

rajatagarwal-ibm commented:
PR https://github.com/terraform-ibm-modules/documentation/pull/48
